dataStructures/queues/queue.go

```diff
@@ -39,52 +39,52 @@ func copyQ(q Queue) Queue {
 	return append(Queue{}, q...)
 }
 
-// Add method takes an order and adds it to the queue based on priority.
-func (q *Queue) Add(order *Order) {
-	if len(*q) == 0 {
-		*q = Queue{order}
-	} else {
-		var appended bool
-
-	QLoop:
-		for i, existingOrder := range *q {
-			var insert int
-
-			switch {
-			case order.priority < existingOrder.priority:
-				insert = i
-			case order.priority == existingOrder.priority:
-				insert = 1 + i
-				// peek to check next order in the queue.
-				if insert < len(*q) && (*q)[insert].priority == order.priority {
-					// keep looking for the best location to insert.
-					continue QLoop
-				}
-			default:
-				continue QLoop
+// binarySearch method will help to pick a slot to fix the next order
+// with complexity of Olog(n)
+func getInsertionSlot(q Queue, value int) int {
+	low := 0
+	high := len(q) - 1
+
+	for low <= high {
+		mid := (high + low) / 2
+
+		switch {
+		case q[mid].priority == value:
+			if (mid+1) < len(q) && q[mid+1].priority == value {
```

Comment:
```
if i := mid+1; i<len(q) && q[i].priority == value {
  low = i
} else {
  return i
}
```

Makes it easier for someone reading the code.

Reply:
Nice! Thanks